vm/src/builtins/iter.rs

```diff
@@ -13,6 +14,88 @@ use crate::{
     TypeProtocol,
 };
 
+#[derive(Debug)]
+pub struct PositionIterInternal {
+    pub position: AtomicCell<usize>,
+    /// object or PyNone if exhausted
+    pub obj: PyRwLock<PyObjectRef>,
```

Comment:
Replace with PyRwLock<Option<PyObjectRef>> or this is fine?

Reply:
I think it would be beneficial for it to be generic over `<T: PyValue>` so that you don't need to downcast, and then it would make sense for it to be Option

Reply:
Or maybe something like `AsRef<PyObjectRef>` to allow `PyRef<T>` or `PyObjectRef`. I think that *would* require a couple AsRef impls though.